testng-core/src/test/kotlin/org/testng/internal/ClassHelperTest.kt

```diff
@@ -0,0 +1,153 @@
+        val methods = ClassHelper.getAvailableMethods(whichClass.java)
+            .filter {
+                //jacocoInit() is something that Gradle adds. So filter it out.
+                !"\$jacocoInit".contentEquals(it.name)
+            }.toList()
+        val size = expected.size
+        assertThat(methods.size)
+            .withFailMessage("Number of methods found should have been at-least $size")
+            .isEqualTo(size)
+        val actual = methods.map {
+            it.name
+        }.toList()
+        assertThat(actual).containsAll(expected)
+    }
+
+    @DataProvider(name = "scenariosData")
+    fun scenariosData(): Array<Array<Any>> {
+        return arrayOf(
+            arrayOf("AllAvailableMethods", getExpected(), LittlePandaSample::class),
+            arrayOf(
+                "AvailableMethodsWhenOverridingIsInvolved",
+                getExpected("equals", "hashCode", "toString"),
+                BabyPandaSample::class
+            ),
+        )
+    }
+
+    @Test
+    fun testNoClassDefFoundError() {
+        val urlClassLoader = object : URLClassLoader(emptyArray<URL>()) {
+            override fun loadClass(className: String?): Class<*> {
+                throw NoClassDefFoundError()
+            }
+        }
+        ClassHelper.addClassLoader(urlClassLoader)
+        val fakeClassName = UUID.randomUUID().toString()
+        assertThat(ClassHelper.forName(fakeClassName))
+            .withFailMessage("The result should be null; no exception should be thrown.")
+            .isNull()
+    }
+
+    @Test(dataProvider = "data")
+    fun testWithDefaultMethodsBeingOverridden(
+        cls: KClass<*>, expectedCount: Int, vararg expected: String
+    ) {
+        val methods = ClassHelper.getAvailableMethodsExcludingDefaults(cls.java)
+            .map {
+                it.declaringClass.name + "." + it.name
+            }
+        assertThat(methods).hasSize(expectedCount)
+        assertThat(methods).contains(*expected)
+    }
+
+    @DataProvider(name = "data")
+    fun getTestData(): Array<Array<Any>> {
+        return arrayOf(
+            arrayOf(MickJaggerSample::class, 1, MickJaggerSample::class.java.name + ".dance"),
+            arrayOf(
+                JohnTravoltaMovesSample::class,
+                2,
+                JohnTravoltaMovesSample::class.java.name + ".walk",
+                AbstractMovesSample::class.java.name + ".dance"
+            ),
+            arrayOf(
+                BatmanSample::class,
+                3,
+                BatmanSample::class.java.name + ".fly",
+                BatmanSample::class.java.name + ".liftWeights",
+                BatmanSample::class.java.name + ".yellSlogan"
+            )
+        )
+    }
+
+    private fun XmlSuite.addXmlTest(testName: String, vararg clazz: KClass<*>): XmlSuite {
+        XmlTest(this).apply {
+            name = testName
+            xmlClasses = mutableListOf<XmlClass?>().apply {
```

Comment:
why not `xmlClasses = newXmlClass(*clazz)` ?

Reply:
fixed